core/src/Kokkos_View.hpp

```diff
@@ -1497,7 +1557,8 @@ class View : public ViewTraits<DataType, Properties...> {
       const Impl::ViewCtorProp<P...>& arg_prop,
       typename std::enable_if<!Impl::ViewCtorProp<P...>::has_pointer,
                               typename traits::array_layout>::type const&
-          arg_layout)
+          arg_layout,
+      int check_input_args = 0)
```

Comment:
is this ever called in a context where we don't want the check?

Reply:
Yes. TestViewLayoutStrideAssignment.hpp:638 would fail when including the check

Reply:
I've added an enum, good point Damien. Daniel, we need to disable the check for Views constructions coming from some other funky places like LayoutStride_mirror (TestViewLayoutStrideAssignment.hpp:638).